packages/itwin/grouping-mapping-widget/src/widget/components/provider/CustomUIProvider.ts

```diff
@@ -0,0 +1,45 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type { ReactElement } from "react";
+
+/**
+ * Custom UI provider type definition
+ */
+export interface CustomUIProvider {
+  /**
+   * Unique identifier of the UI provider
+   */
+  name: string;
+  /**
+   * Display label in the widget
+   */
+  displayLabel: string;
+  /**
+   * Custom UI Component to build query interactively. Refer to SearchUIProvider/ManualUIProvider.
+   */
+  uiComponent: (props: CustomUIProviderProps) => JSX.Element;
+  /**
+   * Optional icon, will be shown before display label in widget
+   */
+  icon?: ReactElement;
+}
+
+/**
+ * Custom UI Provider props
+ */
+export interface CustomUIProviderProps {
```

Comment:
I think these props are not for the provider, but for the `uiComponent` itself. The name should be changed.